tests/background/tab-state-test.js

```diff
@@ -152,12 +156,107 @@ describe('TabState', () => {
       clock.restore();
     });
 
+    it("doesn't query the service for invalid URLs", async () => {
+      const testValue = 42;
+      getStub.resolves(testValue);
+      uriForBadgeRequestStub.throws('any error');
+      const tabState = new TabState({ 1: { state: states.ACTIVE } });
+
+      const promise = tabState.updateAnnotationCount(1, 'invalidOrblocked');
+      clock.tick(INITIAL_WAIT_MS);
+
+      await promise;
+      assert.notCalled(getStub);
+      assert.equal(tabState.getState(1).annotationCount, 0);
+    });
+
+    it('updates the annotationCount (immediately) if previous URL query is still in the cache', async () => {
+      uriForBadgeRequestStub.returnsArg(0);
```

Comment:
If the same initialization is repeated for a stub in many places you can set it as a "default" behavior in the `beforeEach` block or other function that creates the stub, with something like: `uriForBadgeRequestStub = sinon.stub().returnsArg(0)`.